test/common/event/dispatcher_impl_test.cc

```diff
@@ -85,7 +85,7 @@ class DispatcherImplTest : public testing::Test {
     dispatcher_thread_->join();
   }
 
-  NiceMock<Stats::MockStore> scope_;
+  NiceMock<Stats::MockStore> scope_; // Used in InitializeStats, needs to outlive dispatcher_.
```

Comment:
Can you clarify that it's not about outliving `dispatcher_` object lifetime (which was true even before thie PR), but the `dispatcher_->exit()`.